utils/printTracks.py

```diff
@@ -24,14 +24,26 @@ def printTracks(tracksDir, filename=None):
             numStops = len(data['stops']['values'])
 
             speedLimitValues = [v[1]*(3.6 if data['speed limits']['units']['velocity'] == 'm/s' else 1) for v in data['speed limits']['values']]  # km/h
-            gradientValues = [v[1] for v in data['gradients']['values']]  # permil
             speedLimitPositions = [v[0]*(1e3 if data['speed limits']['units']['position'] == 'km' else 1) for v in data['speed limits']['values']]  # m
-            gradientPositions = [v[0]*(1e3 if data['gradients']['units']['position'] == 'km' else 1) for v in data['gradients']['values']]  # m
 
-            positions = sorted(set(speedLimitPositions + gradientPositions + [length]))
+            if 'gradient' in data:
+                gradientValues = [v[1] for v in data['gradients']['values']]  # permil
```

Comment:
I am being pedantic here but please introduce empty lines in the nested if statements (and try/except) as in the rest of the code